src/draw/opengles/lv_draw_opengles.c

```diff
@@ -0,0 +1,666 @@
+            GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST));
+            GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE));
+            GL_CALL(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE));
+            GL_CALL(glPixelStorei(GL_UNPACK_ALIGNMENT, 1));
+
+            /*Color depth: 8 (L8), 16 (RGB565), 24 (RGB888), 32 (XRGB8888)*/
+#if LV_COLOR_DEPTH == 8
+            GL_CALL(glTexImage2D(GL_TEXTURE_2D, 0, GL_R8, w, h, 0, GL_RED, GL_UNSIGNED_BYTE, NULL));
+#elif LV_COLOR_DEPTH == 16
+            GL_CALL(glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB565, w, h, 0, GL_RGB, GL_UNSIGNED_SHORT_5_6_5, NULL));
+#elif LV_COLOR_DEPTH == 24
+            GL_CALL(glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB, w, h, 0, GL_BGR, GL_UNSIGNED_BYTE, NULL));
+#elif LV_COLOR_DEPTH == 32
+            GL_CALL(glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, w, h, 0, GL_BGRA, GL_UNSIGNED_BYTE, NULL));
+#else
+#error("Unsupported color format")
+#endif
+
+            glGenerateMipmap(GL_TEXTURE_2D);
+
+            layer->user_data = (void *)(uintptr_t)texture;
+        }
+        else {
+            layer->user_data = (void *)(uintptr_t)lv_opengles_texture_get_texture_id(disp);
+        }
+    }
+
+    t->state = LV_DRAW_TASK_STATE_IN_PROGRESS;
+    draw_opengles_unit->base_unit.target_layer = layer;
+    draw_opengles_unit->base_unit.clip_area = &t->clip_area;
+    draw_opengles_unit->task_act = t;
+
+    execute_drawing(draw_opengles_unit);
+
+    draw_opengles_unit->task_act->state = LV_DRAW_TASK_STATE_READY;
+    draw_opengles_unit->task_act = NULL;
+
+    /*The draw unit is free now. Request a new dispatching as it can get a new task*/
+    lv_draw_dispatch_request();
+    return 1;
+}
+
+static int32_t evaluate(lv_draw_unit_t * draw_unit, lv_draw_task_t * task)
+{
+    LV_UNUSED(draw_unit);
+
+    if(((lv_draw_dsc_base_t *)task->draw_dsc)->user_data == NULL) {
+        task->preference_score = 0;
+        task->preferred_draw_unit_id = DRAW_UNIT_ID_OPENGLES;
+    }
+    return 0;
+}
+
+static bool draw_to_texture(lv_draw_opengles_unit_t * u, cache_data_t * cache_data)
+{
+    lv_draw_task_t * task = u->task_act;
+
+    lv_layer_t dest_layer;
+    lv_memzero(&dest_layer, sizeof(dest_layer));
+
+    int32_t texture_w = lv_area_get_width(&task->_real_area);
+    int32_t texture_h = lv_area_get_height(&task->_real_area);
+
+    lv_draw_buf_t draw_buf;
+    dest_layer.draw_buf = &draw_buf;
+    lv_draw_buf_init(dest_layer.draw_buf, texture_w, texture_h,
+                     LV_COLOR_FORMAT_ARGB8888, LV_STRIDE_AUTO, opengles_render_buf, sizeof(opengles_render_buf));
+    dest_layer.color_format = LV_COLOR_FORMAT_ARGB8888;
+
+    dest_layer.buf_area = task->_real_area;
+    dest_layer._clip_area = task->_real_area;
+    dest_layer.phy_clip_area = task->_real_area;
+    lv_memzero(opengles_render_buf, lv_area_get_size(&dest_layer.buf_area) * 4 + 100);
```

Comment:
magic number +100 bytes allocated for the render buffer?

Reply:
I think it's coming from the SDL renderer. Probably not required, just a leftover.

Reply:
Correct, it came from the SDL renderer. I was trying to leave working things untouched until the end.